pandas/tests/io/test_stata.py

```diff
@@ -1327,3 +1329,22 @@ def test_set_index(self):
             df.to_stata(path)
             reread = pd.read_stata(path, index_col='index')
         tm.assert_frame_equal(df, reread)
+
+    @pytest.mark.parametrize(
+        'column', ['ms', 'day', 'week', 'month', 'qtr', 'half', 'yr'])
+    def test_date_parsing_ignores_format_details(self, column):
+        # GH 17797
+        #
+        # Test that display formats are ignored when determining if a numeric
+        # column is a date value.
+        #
+        # All date types are stored as numbers and format associated with the
+        # column denotes both the type of the date and the display format.
+        #
+        # STATA supports 9 date types which each have distinct units. We test 7
+        # of the 9 types, ignoring %tC and %tb. %tC is a variant of %tc that
+        # accounts for leap seconds and %tb relies on STATAs business calendar.
+        df = read_stata(self.stata_dates)
+        unformatted = df.loc[0, column]
+        formatted = df.loc[0, column + "_fmt"]
+        assert unformatted == formatted
```

Comment:
are these supposed to be ``datetime64[ns]`` dtype?

what happens for the ignored formats? should raise?

Reply:
> are these supposed to be `datetime64[ns]` dtype?

At this point in the code `formatted` and `unformatted` are `pandas._libs.tslib.Timestamp` objects. Every column in `df` has a dtype of `datetime64[ns]` 

> what happens for the ignored formats? should raise?

Ignored formats are not converted to dates (consistent with previous behavior) [source](https://github.com/pandas-dev/pandas/blob/master/pandas/io/stata.py#L1581-L1582)